src/providers/spatialite/qgsspatialiteprovider.cpp

```diff
@@ -5742,7 +5742,18 @@ bool QgsSpatiaLiteProvider::getTableSummaryAbstractInterface( gaiaVectorLayerPtr
   {
     mLayerExtent.set( lyr->ExtentInfos->MinX, lyr->ExtentInfos->MinY,
                       lyr->ExtentInfos->MaxX, lyr->ExtentInfos->MaxY );
-    mNumberFeatures = lyr->ExtentInfos->Count;
+    // This can be wrong! see: GH #29264
+    // mNumberFeatures = lyr->ExtentInfos->Count;
+    sqlite3_database_unique_ptr slPtr;
+    slPtr.reset( sqliteHandle() );
+    int resultCode;
+    sqlite3_statement_unique_ptr stmt { slPtr.prepare( QStringLiteral( "SELECT COUNT(1) FROM %2" ).arg( mQuery ), resultCode )};
+    if ( resultCode == SQLITE_OK )
+    {
+      stmt.step();
+      mNumberFeatures = sqlite3_column_int64( stmt.get(), 0 );
+    }
+    slPtr.release();
```

Comment:
Doesn't this leak?

Reply:
I don't think so

Reply:
What takes ownership of the released pointer?

Reply:
Nothing takes it because the provider owns the pointer in the first place: https://github.com/qgis/QGIS/blob/9d040bb917407e3c936f3a04162eff2b1dcaaad1/src/providers/spatialite/qgsspatialiteprovider.cpp#L1124

Reply:
In that case I don't think it's suitable for storing in a unique_ptr -- why not just use a raw pointer?

Reply:
because of the convenience method `prepare` that come with `sqlite3_database_unique_ptr` and because it returns `sqlite3_statement_unique_ptr` (that is used as a real unique ptr in this code).

Reply:
I've added two comments to clarify ownership.

Reply:
Ok -- I think in future we should consider making mSqliteHandle a sqlite3_database_unique_ptr, but it looks like that gets complicated due to the QgsSqliteHandle class used here. ...